### DISCUSSION TITLE
ACAP artpec-8 Permission issue - Blocked

### DISCUSSION THREAD

**Discussion by rafiomeon** [AxisCommunications/discussions#51] [ACAP] [answered]
Hi,

We have created the ACAP application for artpec-8 using the Docker provided by Axis. We have compiled our custom video analytics application which was already running in Q1615 using artpec-7. However we are facing an issue in artpec8 where the eap is installed in P3267. Our services are not getting launched due to Permission issue. We didnt have this permission issue with previous acap build running with artpec-7 mode in q1615. 

Docker Used to build artpec8 acap:
https://github.com/AxisCommunications/docker-acap

Could you let us know how to resolve this Permission issue in acap while running in P3267 with artpec-8 . 

Thanks,
![IssuePermission](https://user-images.githubusercontent.com/67184931/188898936-bac38d1b-5e6d-4b9e-bc9d-1896053a9704.jpg)
![acap running](https://user-images.githubusercontent.com/67184931/188899915-448fb10a-865b-411b-9244-4b7feab1884d.jpg)

Rafi

**Comment by n8eye200** [comment, marked as answer]
Hey @rafiomeon !

I ran into something similar when putting a script on the SD card, due to the fact that the "noexec" flag is set on the drive.  My hunch is that you are experiencing this, too.

To check, try running the mount command -

```
root@axis-b8REDACTED:~# mount
/dev/dm-0 on / type squashfs (ro,relatime,errors=continue)
proc on /proc type proc (rw,relatime)
sysfs on /sys type sysfs (rw,relatime)
  ...  (Removed a bunch of boring output here...)
/dev/mmcblk1p1 on /var/volatile/spool/storage/SD_DISK type ext4 (rw,nosuid,nodev,noexec,noatime,nodiratime,lazytime,commit=1,data=ordered)
  ...  (And here...)
```

Note the noexec stuff in parenthesis behind the mount point that you are running the script from.

To get around this temporarily to see if everything else in your solution is running happily, you can re-mount the filesystem with the noexec flag removed, like so -

```
root@axis-b8REDACTED:~# mount -o remount,exec /var/volatile/spool/storage/SD_DISK
```

When you run the mount command again, you'll see that the flag is no longer set and you should be able to execute your script.

Note that

1. This isn't exactly a great solution, noexec is there to keep things safe
2. When you reboot your device, the filesystem will again be mounted with the noexec flag set

I am still looking for a long-term solution to this noexec thing, and will add to this discussion if something good pops up.  If anything, I hope this info will help you diagnose what is going on on your end!

**Reply by rafiomeon** [reply]
Thanks @n8eye200 .. This solution worked for us in resolving the permission issue. But still looking for long-term solution.
